tools/serverpod_cli/lib/src/generator/dart/library_generators/entities_library_generator.dart

```diff
@@ -1326,41 +1343,157 @@ class SerializableEntityLibraryGenerator {
+                    'createTable': Method(
+                      (m) => m
+                        ..requiredParameters.addAll([
+                          Parameter((p) => p..name = 'foreignTableRelation'),
+                        ])
+                        ..lambda = true
+                        ..body = field.type
+                            .reference(
+                          serverCode,
+                          subDirParts: classDefinition.subDirParts,
+                          config: config,
+                          nullable: false,
+                          typeSuffix: 'Table',
+                        )
+                            .call([], {
+                          'tableRelation': refer('foreignTableRelation')
+                        }).code,
+                    ).closure
+                  },
+                ),
+              )
+              .statement,
+          Code('return _${field.name}!;'),
+        ])));
+    }
+
+    return getters;
+  }
+
+  List<Method> _buildEntityTableClassManyRelationGetters(
+    List<SerializableEntityFieldDefinition> fields,
+    ClassDefinition classDefinition,
+  ) {
+    List<Method> getters = [];
+
+    var manyRelationFields =
+        fields.where((f) => f.relation is ListRelationDefinition);
+
+    for (var field in manyRelationFields) {
+      var listRelation = field.relation as ListRelationDefinition;
+
+      getters.add(Method((m) => m
+        ..name = field.name
+        ..type = MethodType.getter
+        ..returns = TypeReference((t) => t
+          ..symbol = 'ManyRelation'
+          ..url = serverpodUrl(serverCode)
+          ..types.add(field.type.generics.first.reference(
+            serverCode,
+            subDirParts: classDefinition.subDirParts,
+            config: config,
+            nullable: false,
+            typeSuffix: 'Table',
+          )))
+        ..body = Block.of([
+          Code('if (_${field.name} != null) return _${field.name}!;'),
+          declareVar('relationTable')
+              .assign(
+                refer(
+                  'createRelationTable',
+                  'package:serverpod/serverpod.dart',
+                ).call(
+                  [],
+                  {
+                    'relationFieldName': literalString(field.name),
+                    'field': refer(classDefinition.className)
+                        .property('t')
+                        .property(listRelation.fieldName),
+                    'foreignField': field.type.generics.first
+                        .reference(
+                          serverCode,
+                          subDirParts: classDefinition.subDirParts,
+                          config: config,
+                          nullable: false,
+                        )
+                        .property('t')
+                        .property(
+                          listRelation.implicitForeignField
+                              ? '\$${listRelation.foreignFieldName}'
```

Comment:
Suggestion: Use the util method, I moved that method into its own file called `createHiddenField` or similar.